### PR TITLE
[llvm-objcopy] Fix gap-fill/pad-to tests

### PR DESCRIPTION
**Pull request by quic-akaryaki**
The tests added in PR #65815 fail on Apple buildbot because the `od` printed addresses have a different number of leading zeroes. Mask leading zeroes with a regex.

### CODE REVIEW

**Review by jansvoboda11 — APPROVED** [review]
LGTM, thanks!

File: llvm/test/tools/llvm-objcopy/ELF/gap-fill.test [hunk 1]
```diff
@@ -24,65 +24,65 @@
 ## Test no gap fill with all allocatable output sections.
 # RUN: llvm-objcopy -O binary %t %t-default
 # RUN: od -v -Ax -t x1 %t-default | FileCheck %s --check-prefix=DEFAULT --match-full-lines
```
Comment (MaskRay) [inline llvm/test/tools/llvm-objcopy/ELF/gap-fill.test:26]: We may need `--ignore-case` to support z/OS

**Review by MaskRay — APPROVED** [review]
We also need --ignore-case

**Comment by quic-akaryaki** [comment]
The build failed due to an unrelated reason (Flang :: Driver/ctofortran.f90). If there no objections, I would like to merge this fix.

**Pull request merged**